modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/content/ContentManagerBean.java

```diff
@@ -626,6 +628,76 @@ public void completeDeployPackageRequest(DeployPackagesResponse response) {
             entityManager.persist(history);
             persistedRequest.addInstalledPackageHistory(history);
         }
+
+        // Clean old version bits for packages..
+
+        for (DeployIndividualPackageResponse singleResponse : response.getPackageResponses()) {
+            PackageDetailsKey key = singleResponse.getKey();
+            Query packageVersionQuery = entityManager
+                    .createNamedQuery(PackageVersion.QUERY_FIND_BY_PACKAGE_DETAILS_KEY_WITH_NON_NULL_RESOURCE_TYPE);
+            packageVersionQuery.setParameter("packageName", key.getName());
+            packageVersionQuery.setParameter("packageTypeName", key.getPackageTypeName());
+            packageVersionQuery.setParameter("architectureName", key.getArchitectureName());
+            packageVersionQuery.setParameter("version", key.getVersion());
+            packageVersionQuery.setParameter("resourceTypeId", resourceTypeId);
+            PackageVersion packageVersion = (PackageVersion) packageVersionQuery.getSingleResult();
+            purgePackageBits(packageVersion.getGeneralPackage().getId());
+        }
+
+        deleteOrphanedBlobs();
+    }
+
+    public void removeHistoryDeploymentsBits(){
```

Comment:
I feel this method should require authentication. In its current form, anyone is able to remove the deployment bits.

Although this is probably fine for this release as this data is unreachable and unusable, my concern is if we change this in the future. It seems that there should be some protection for this method that requires 

Also not sure how easy the permission check will be. Perhaps this will be a super-user function? Otherwise, the user will have to have the manage resource permission for the resource the history is being removed from.